### PR TITLE
Ports: Set `$LD` for Clang, GCC and host in `.hosted_defs.sh`

### PR DESCRIPTION
**Pull request by EWouters**
Sets `$LD` to `$HOST_LD` in `.port_include.sh` if it is defined as well,
else it will be set to `ld`.

Makes libiconv build with the Clang toolchain.

This also impacts other ports when building them with the Clang
toolchain, and might result in more ports building correctly.

### CODE REVIEW

**Comment by timschumi** [comment]
We should probably add a definition for `LD` to all three things, right? GCC, Clang,  and host.

**Comment by BertalanD** [comment]
Commit message has a typo: `ld.lld`.

File: Ports/.hosted_defs.sh [hunk 1]
```diff
@@ -27,6 +27,7 @@ if [ "$SERENITY_TOOLCHAIN" = "Clang" ]; then
     export SERENITY_TOOLCHAIN_BINDIR="${SERENITY_SOURCE_DIR}/Toolchain/Local/clang/bin"
     export CC="${SERENITY_ARCH}-pc-serenity-clang"
     export CXX="${SERENITY_ARCH}-pc-serenity-clang++"
+    export LD="${SERENITY_TOOLCHAIN_BINDIR}/ld.lld"
```
Comment (ADKaster) [inline Ports/.hosted_defs.sh:30]: Just keeping this exported even when we do a `host_env` will probably break on macOS. 

We should probably `export HOST_LD="${LD:=}"` to default it to 'not set'. Normally you wouldn't set that variable for just normal host programming.

Comment (EWouters) [reply Ports/.hosted_defs.sh:31]: Thank you, as discussed on Discord, I've set it to `ld` if it was not defined.

File: Ports/.hosted_defs.sh [hunk 2]
```diff
@@ -38,6 +40,7 @@ else
     export SERENITY_TOOLCHAIN_BINDIR="${SERENITY_SOURCE_DIR}/Toolchain/Local/${SERENITY_ARCH}/bin"
     export CC="${SERENITY_ARCH}-pc-serenity-gcc"
     export CXX="${SERENITY_ARCH}-pc-serenity-g++"
+    export LD="${SERENITY_TOOLCHAIN_BINDIR}/x86_64-pc-serenity-ld"
```
Comment (timschumi) [inline Ports/.hosted_defs.sh:43]: Hardcoded arch.

Comment (timschumi) [inline Ports/.hosted_defs.sh:43]: Unlike the other entries, does `${SERENITY_ARCH}-pc-serenity-ld` not work here? It feels unnecessary to forego PATH resolving manually.

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Ports/.hosted_defs.sh (hunk 2 above)
Comment (EWouters) [reply Ports/.hosted_defs.sh]: Right, thank you.

**Review by ADKaster — APPROVED** [review]

**Pull request merged**